sormas-ui/src/main/java/de/symeda/sormas/ui/externalsurveillanceservice/ExternalSurveillanceShareComponent.java

```diff
@@ -89,24 +93,39 @@ private HorizontalLayout createHeader(
 			headerLayout.setExpandRatio(sendButton, 1);
 			headerLayout.setComponentAlignment(sendButton, Alignment.MIDDLE_RIGHT);
 		}
-		if (deleteHandler != null && isVisibleDeleteButton(caseUuid)) {
-			Button deleteButton = ButtonHelper.createIconButton("", VaadinIcons.TRASH, e -> deleteHandler.run(), ValoTheme.BUTTON_ICON_ONLY);
-			headerLayout.addComponent(deleteButton);
-			headerLayout.setComponentAlignment(deleteButton, Alignment.MIDDLE_RIGHT);
-			headerLayout.setWidth(100, Unit.PERCENTAGE);
+
+		if (caseUuid != null) {
+			if (deleteHandler != null && isVisibleDeleteButtonForCase(caseUuid)) {
+				addDeleteButtonToLayout(deleteHandler, headerLayout);
+			}
+		}
+
+		if (eventUuid != null) {
+			if (deleteHandler != null && isVisibleDeleteButtonForEvent(eventUuid)) {
+				addDeleteButtonToLayout(deleteHandler, headerLayout);
+			}
 		}
 
 		return headerLayout;
 	}
 
-	private boolean isVisibleDeleteButton(String caseUuid) {
+	private void addDeleteButtonToLayout(Runnable deleteHandler, HorizontalLayout headerLayout) {
+		Button deleteButton = ButtonHelper.createIconButton("", VaadinIcons.TRASH, e -> deleteHandler.run(), ValoTheme.BUTTON_ICON_ONLY);
+		headerLayout.addComponent(deleteButton);
+		headerLayout.setComponentAlignment(deleteButton, Alignment.MIDDLE_RIGHT);
+		headerLayout.setWidth(100, Unit.PERCENTAGE);
+	}
+
+	private boolean isVisibleDeleteButtonForCase(String caseUuid) {
 		if (caseUuid != null) {
-			CaseDataDto caseDataDto = FacadeProvider.getCaseFacade().getCaseDataByUuid(caseUuid);
-			boolean isSharedCase = FacadeProvider.getExternalShareInfoFacade().isSharedCase(caseUuid);
+			return FacadeProvider.getExternalShareInfoFacade().isSharedCase(caseUuid);
+		}
+		return true;
```

Comment:
caseUuid is checked for null twice this return will not be reached in the current implementation